### PR TITLE
Adjust ALS .js.map files for debugging

### PR DESCRIPTION
**Pull request by TamiTakamiya**
When `npm run compile` copies ALS js/d.ts/js.map files from packages/ansible-language-server to out directory, we need to adjust the source path in .js.map files so that we can set breakpoints on TypeScript source files in debugging.

### CODE REVIEW

**Review by ssbarnea — CHANGES_REQUESTED** [review]
@TamiTakamiya Can we please find another way to address this issue? -- one that does not adds extra copy-files or replace-text-in-files?

With the tsconfig consolidated by https://github.com/ansible/vscode-ansible/pull/1279 it should be much easier to reconfigure how we compile to make it work "by default" and without any tricks.

PS. I would prefer to make use of `out/` folder, so we do not spam our source tree.

**Comment by TamiTakamiya** [comment]
> @TamiTakamiya Can we please find another way to address this issue? -- one that does not adds extra copy-files or replace-text-in-files?
> 
> With the tsconfig consolidated by #1279 it should be much easier to reconfigure how we compile to make it work "by default" and without any tricks.
> 
> PS. I would prefer to make use of `out/` folder, so we do not spam our source tree.

Ok, I will consider other options.

**Comment by TamiTakamiya** [comment]
> > @TamiTakamiya Can we please find another way to address this issue? -- one that does not adds extra copy-files or replace-text-in-files?
> > With the tsconfig consolidated by #1279 it should be much easier to reconfigure how we compile to make it work "by default" and without any tricks.
> > PS. I would prefer to make use of `out/` folder, so we do not spam our source tree.
> 
> Ok, I will consider other options.

Another related-issue is that webpack generates a "bundled" server.js file in the `out` directory. 
```
~/git/ansible/vscode-ansible[TamiTakamiya/fix-als-js-map] $ yarn compile
~/git/ansible/vscode-ansible[TamiTakamiya/fix-als-js-map] $ ls -l out/server/src/server.js
-rw-r--r--. 1 ttakamiy ttakamiy 2032 Apr 25 10:48 out/server/src/server.js
~/git/ansible/vscode-ansible[TamiTakamiya/fix-als-js-map] $ yarn webpack
webpack compiled successfully

webpack compiled successfully

webpack compiled successfully

webpack compiled successfully

webpack compiled successfully

webpack compiled successfully

webpack compiled successfully

webpack compiled successfully
~/git/ansible/vscode-ansible[TamiTakamiya/fix-als-js-map] $ ls -l out/server/src/server.js
-rw-r--r--. 1 ttakamiy ttakamiy 823716 Apr 25 10:48 out/server/src/server.js
~/git/ansible/vscode-ansible[TamiTakamiya/fix-als-js-map] $ yarn compile
~/git/ansible/vscode-ansible[TamiTakamiya/fix-als-js-map] $ ls -l out/server/src/server.js
-rw-r--r--. 1 ttakamiy ttakamiy 2032 Apr 25 10:48 out/server/src/server.js
~/git/ansible/vscode-ansible[TamiTakamiya/fix-als-js-map] $
```

I could debug ALS by running `yarn compile` after `yarn webpack`, but it may not be ideal. I think I also need to take a look at the options for webpack.

**Comment by TamiTakamiya** [comment]
@ssbarnea It turned out to be a one-line change in the `webpack.config.ts` file. With the change, we no longer need to copy ALS js files by ourselves and the local build process looks much simpler than before. Thank you for your comment!

File: yarn.lock [hunk 1]
```diff
@@ -3250,24 +3249,6 @@ __metadata:
   languageName: node
   linkType: hard
 
-"copyfiles@npm:^2.4.1":
```
Comment (TamiTakamiya) [inline yarn.lock:3253]: copyfiles module, which was used for copying ALS .js files to `out/server` directory is no longer needed.

**Comment by TamiTakamiya** [comment]
The "bundled" js seems to cause lots of noises from external libraries in code coverage report. Some changes are needed for suppressing them.
image

**Comment by TamiTakamiya** [comment]
I have updated `package.json` again not to use webpack's bundleed js in testing so that we could get cleaner code coverage.

**Review by priyamsahoo — APPROVED** [review]

**Comment by priyamsahoo** [comment]
@TamiTakamiya, could you update the documentation on setting up the dev environment and running the extension with breakpoints?

**Review by ssbarnea — APPROVED** [review]

**Comment by TamiTakamiya** [comment]
> @TamiTakamiya, could you update the documentation on setting up the dev environment and running the extension with breakpoints?

@priyamsahoo I think this [development.md](https://github.com/ansible/vscode-ansible/blob/main/docs/development.md) is the one to be updated. I looked at the document and found it was still assuming ansible-language-server as a separate repo.  Also it assumes the use of the `task` tool. I tried to set it up on my Fedora 39 workstastion, but running the default task failed with:
```
ERROR:   dbus-daemon was not detecting as running and that would interfere with testing (xvfb).
task: Failed to run task "default": exit status 55
```
I think xvfb is not required for local development and some changes would be necessary in `Taskfile.yml` for local development without breaking CI. I think it would take some time and I opened #1281 for updating development docs (and possbly some additional work).

Even though I want to update the doc with #1281, following are the steps I am using for debugging extension/language server:

- Setup
    1. `yarn install`
    2. `yarn webpack` -- this cleans out directries both for ALS & extension, compile sources then build webpack
    
- Debug extension with breakpoints
    1. Start a debug session with `Launch Extension`
    
- Debug language server with breakpoints
    1. Start a debug session with `Launch Extension`, then
    1. Start another debug session with `Attach to Server`
    
The drawback is the processing time as building webpack takes more time than just running tsc.  Even though the original solution in this PR did not have this as it manipulated .js.map files directly and let non-bundled js files for debugging, I believe using the bundled js would be less confusing.

Note: Non-bundled js files are still used in e2e testing for providing cleander code coverage reports.

**Comment by TamiTakamiya** [comment]
ci / test (macos) fails constantly with errors like
```
  ERROR in /Users/runner/work/vscode-ansible/vscode-ansible/src/extension.ts
  ./src/extension.ts 5:30-48
  [tsl] ERROR in /Users/runner/work/vscode-ansible/vscode-ansible/src/extension.ts(5,31)
        TS6059: File '/Users/runner/work/vscode-ansible/vscode-ansible/src/features/vault.ts' is not under 'rootDir' '/Users/runner/work/vscode-ansible/vscode-ansible/packages/ansible-language-server'. 'rootDir' is expected to contain all source files.
    The file is in the program because:
      Imported via "./features/vault" from file '/Users/runner/work/vscode-ansible/vscode-ansible/src/extension.ts'
      Root file specified for compilation
  task: Failed to run task "package": exit status 1
  ts-loader-default_02839d45964937a4
```
For some reasons, packaging on macos does not seem to work with the bundled js... I will take a look.

**Comment by TamiTakamiya** [comment]
> ci / test (macos) fails constantly with errors like
> 
> ```
>   ERROR in /Users/runner/work/vscode-ansible/vscode-ansible/src/extension.ts
>   ./src/extension.ts 5:30-48
>   [tsl] ERROR in /Users/runner/work/vscode-ansible/vscode-ansible/src/extension.ts(5,31)
>         TS6059: File '/Users/runner/work/vscode-ansible/vscode-ansible/src/features/vault.ts' is not under 'rootDir' '/Users/runner/work/vscode-ansible/vscode-ansible/packages/ansible-language-server'. 'rootDir' is expected to contain all source files.
>     The file is in the program because:
>       Imported via "./features/vault" from file '/Users/runner/work/vscode-ansible/vscode-ansible/src/extension.ts'
>       Root file specified for compilation
>   task: Failed to run task "package": exit status 1
>   ts-loader-default_02839d45964937a4
> ```
> 
> For some reasons, packaging on macos does not seem to work with the bundled js... I will take a look.

After rerunning the job, it succeeded.  Let me run few more tests.

**Comment by TamiTakamiya** [comment]
> > ci / test (macos) fails constantly with errors like
> > ```
> >   ERROR in /Users/runner/work/vscode-ansible/vscode-ansible/src/extension.ts
> >   ./src/extension.ts 5:30-48
> >   [tsl] ERROR in /Users/runner/work/vscode-ansible/vscode-ansible/src/extension.ts(5,31)
> >         TS6059: File '/Users/runner/work/vscode-ansible/vscode-ansible/src/features/vault.ts' is not under 'rootDir' '/Users/runner/work/vscode-ansible/vscode-ansible/packages/ansible-language-server'. 'rootDir' is expected to contain all source files.
> >     The file is in the program because:
> >       Imported via "./features/vault" from file '/Users/runner/work/vscode-ansible/vscode-ansible/src/extension.ts'
> >       Root file specified for compilation
> >   task: Failed to run task "package": exit status 1
> >   ts-loader-default_02839d45964937a4
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > For some reasons, packaging on macos does not seem to work with the bundled js... I will take a look.
> 
> After rerunning the job, it succeeded. Let me run few more tests.

I re-ran the test a couple of times, but could not recreate the issue.  The log indicates there seem some timings that webpack set's its root dir to ALS's then attempts to compile extension's TypeScript code. Though I am going to merge this to main, I'll keep watching ci jobs if those issues re-appear.

**Pull request merged**